standalone/src/test/java/io/pravega/local/InProcPravegaClusterTest.java

```diff
@@ -118,69 +44,7 @@ ClientConfig prepareValidClientConfig() {
      *
      */
     @Test(timeout = 30000)
-    public void testWriteAndReadEventWithValidClientConfig() throws ExecutionException,
-            InterruptedException, ReinitializationRequiredException {
-        String scope = scopeName();
-        String streamName = streamName();
-        int numSegments = 1;
-        String message = eventMessage();
-
-        ClientConfig clientConfig = prepareValidClientConfig();
-
-        @Cleanup
-        StreamManager streamManager = StreamManager.create(clientConfig);
-        assertNotNull(streamManager);
-
-        boolean isScopeCreated = streamManager.createScope(scope);
-        assertTrue("Failed to create scope", isScopeCreated);
-
-        boolean isStreamCreated = streamManager.createStream(scope, streamName, StreamConfiguration.builder()
-                .scalingPolicy(ScalingPolicy.fixed(numSegments))
-                .build());
-        Assert.assertTrue("Failed to create the stream ", isStreamCreated);
-
-        @Cleanup
-        EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(scope, clientConfig);
-
-        // Write an event to the stream.
-
-        @Cleanup
-        EventStreamWriter<String> writer = clientFactory.createEventWriter(streamName,
-                new JavaSerializer<String>(),
-                EventWriterConfig.builder().build());
-        writer.writeEvent(message).get();
-        log.debug("Done writing message '{}' to stream '{} / {}'", message, scope, streamName);
-
-        // Now, read the event from the stream.
-
-        String readerGroup = UUID.randomUUID().toString().replace("-", "");
-        ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder()
-                .stream(Stream.of(scope, streamName))
-                .disableAutomaticCheckpoints()
-                .build();
-
-        @Cleanup
-        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scope, clientConfig);
-        readerGroupManager.createReaderGroup(readerGroup, readerGroupConfig);
-
-        @Cleanup
-        EventStreamReader<String> reader = clientFactory.createReader(
-                "readerId", readerGroup,
-                new JavaSerializer<String>(), ReaderConfig.builder().initialAllocationDelay(0).build());
-
-        // Keeping the read timeout large so that there is ample time for reading the event even in
-        // case of abnormal delays in test environments.
-        String readMessage = reader.readNextEvent(10000).getEvent();
-        log.info("Done reading event [{}]", readMessage);
-
-        assertEquals(message, readMessage);
-    }
-
-
-    @After
-    public void tearDown() throws Exception {
-        if (localPravega != null) {
-            localPravega.close();
-        }
+    public void testWriteAndReadEventWithValidClientConfig() throws Exception {
+        EMULATOR.testWriteAndReadAnEventWithValidClientConfig("TestScope", "TestStream", msg, prepareValidClientConfig());
```

Comment:
Why is this method being moved?

Reply:
refactored the code to move the sanity test to a common class.